src/gtk/webview_webkit2.cpp

```diff
@@ -994,6 +994,38 @@ bool wxWebViewWebKit::SetUserAgent(const wxString& userAgent)
     return true;
 }
 
+bool wxWebViewWebKit::SetProxy(const wxString& proxy)
+{
+#if WEBKIT_CHECK_VERSION(2, 16, 0)
+    if (wx_check_webkit_version(2, 16, 0))
+    {
+        const auto context = static_cast<WebKitWebContext*>(m_config.GetNativeConfiguration());
+        wxCHECK_MSG( context, false, "no context?" );
+
+        const auto data_manager = webkit_web_context_get_website_data_manager(context);
+        wxCHECK_MSG( data_manager, false, "no data manager?" );
+
+        const auto proxy_settings = webkit_network_proxy_settings_new(
+            proxy.utf8_str(),
+            nullptr // no hosts to ignore
+        );
+        wxCHECK_MSG( proxy_settings, false, "failed to create proxy settings" );
+
+        webkit_website_data_manager_set_network_proxy_settings(
+            data_manager,
+            WEBKIT_NETWORK_PROXY_MODE_CUSTOM,
+            proxy_settings
+        );
+
+        webkit_network_proxy_settings_free(proxy_settings);
+    }
+#else // WebKit < 2.16 doesn't support setting proxy
+    wxUnusedVar(proxy);
+#endif // WebKit 2.16+
+
+    return false;
```

Comment:
it seems to return false always

Reply:
Oops, indeed, thanks for noticing this!

Fixed now and I've also added `wxLogError()` calls.